CeeFM/Decideify
# Post Reactions Version 2 are now live!

**Description**
React to posts by clicking the new dropdown menu with all possible emoji reactions listed! Reactions on any given post don't show until they've been used at least once on that post - if they've been used once and then the user removes that post reaction, they disappear right then and there. Same for adding a new one that isn't visible - when someone clicks it from the dropdown, it shows up!

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- On social.js or the /social page you can:
  - Click the new dropdown menu below any post to see a list of all possible post reactions
  - Clicking an option from the list will add the reaction to the post - if it has been used already, the reaction count will be increased by one. If it wasn't added to that post before, then the reaction button will now show with a reaction count of 1.
  - Clicking the button for the reaction you just added to the post will remove your reaction from the post. You can do this directly from the post or by selecting the option from the dropdown menu.
  - If you were the only person who used that reaction on the post, it will disappear from view after you remove it. If you add it again or any other currently hidden reactions, they will appear on the post.

Checklist
